module-bsp/board/rt1051/bsp/magnetometer/magnetometer.cpp

```diff
@@ -118,234 +122,280 @@ namespace bsp
+            if (i2cRead(als31300::CONF_REG, readRegister)) {
+                LOG_DEBUG("CONF register value verification: %" PRIu32,
+                          static_cast<std::uint32_t>(als31300::conf_reg(readRegister)));
             }
             else {
-                LOG_DEBUG("INT is fine, sparing a write");
-            }
-
-            if (timerHandle == nullptr) {
-                timerHandle =
-                    xTimerCreate("SliderTimer", pdMS_TO_TICKS(magnetometerPollIntervalMs), true, nullptr, TimerHandler);
-                if (timerHandle == nullptr) {
-                    LOG_ERROR("Could not create the timer for magnetometer state change detection!");
-                    return kStatus_Fail;
-                }
+                LOG_ERROR("Failed to read from CONF register!");
             }
+        }
+        else {
+            LOG_DEBUG("CONF register unchanged, skipping write");
+        }
 
-            // POWER register
-            i2cRead(als31300::PWR_REG, read_reg);
-            const als31300::pwr_reg current_reg_pwr = read_reg;
-            LOG_DEBUG("POWER read:\t%" PRIu32, static_cast<uint32_t>(current_reg_pwr));
-            als31300::pwr_reg reg_pwr = current_reg_pwr;
-            reg_pwr.I2C_loop_mode     = als31300::PWR_REG_LOOP_MODE_single; // we don't want constant data flow
-            reg_pwr.sleep             = als31300::PWR_REG_SLEEP_MODE_active;
-            reg_pwr.count_max_LP_mode = static_cast<uint8_t>(LPDCM_INACTIVE_TIME::inactive_10ms);
-
-            i2cWrite(als31300::PWR_REG, reg_pwr);
-            LOG_DEBUG("POWER wrote:\t%" PRIu32, static_cast<uint32_t>(reg_pwr));
+        // INTERRUPTS register
+        readStatus                                 = i2cRead(als31300::INT_REG, readRegister);
+        const als31300::int_reg currentIntRegister = readRegister;
+        if (readStatus) {
+            LOG_DEBUG("Current INT register value: %" PRIu32, static_cast<std::uint32_t>(currentIntRegister));
+        }
+        else {
+            LOG_ERROR("Failed to read INT register!");
+        }
 
-            xTimerStart(timerHandle, 1000);
+        als31300::int_reg newIntRegister    = currentIntRegister;
+        newIntRegister.int_eeprom_en        = als31300::INT_REG_INT_EEPROM_disable;
+        newIntRegister.int_mode             = als31300::INT_REG_INT_MODE_threshold;
+        newIntRegister.int_threshold_signed = als31300::INT_REG_THRESHOLD_absolute;
+        newIntRegister.int_X_en             = als31300::INT_REG_INT_CHANNEL_disabled;
+        newIntRegister.int_Y_en             = als31300::INT_REG_INT_CHANNEL_disabled;
+        newIntRegister.int_Z_en             = als31300::INT_REG_INT_CHANNEL_disabled;
+        newIntRegister.int_X_threshold      = 1;
+        newIntRegister.int_Y_threshold      = 4;
+        newIntRegister.int_Z_threshold      = 0;
+
+        if (currentIntRegister != newIntRegister) {
```

Comment:
Changed and also removed the code responsible for reading and printing the value after it was written - the debug level is turned off for this file and the messages are not visible in production code anyway, if someone should need it during debugging it can always be added.

Reply:
Sure, we don't need unreachable logging :)